src/segment_plastic.go

```diff
@@ -0,0 +1,245 @@
+	status += stringIfValue(g.Unmerged, "x")
+	return strings.TrimSpace(status)
+}
+
+func (g *plastic) getPlasticStatus() {
+	output := g.getCmCommandOutput("status", "--all", "--machinereadable")
+	splittedOutput := strings.Split(output, "\n")
+	// compare to head
+	currentChangeset := g.parseStatusChangeset(splittedOutput[0])
+	headChangeset := g.getHeadChangeset()
+	g.Behind = headChangeset > currentChangeset
+
+	// parse file state
+	g.parseFilesStatus(splittedOutput)
+}
+
+func (g *plastic) parseFilesStatus(output []string) {
+	if len(output) <= 1 {
+		return
+	}
+	for _, line := range output[1:] {
+		if len(line) < 3 {
+			continue
+		}
+		if matchString(`(?i)merge\s+from\s+[0-9]+\s*$`, line) {
+			g.Unmerged++
+			continue
+		}
+		code := line[0:2]
+		switch code {
+		case "LD":
+			g.Deleted++
+		case "AD", "PR":
+			g.Added++
+		case "LM":
+			g.Moved++
+		case "CH":
+			g.Modified++
+		}
+	}
+	g.Changed = g.Added > 0 || g.Deleted > 0 || g.Modified > 0 || g.Moved > 0 || g.Unmerged > 0
+}
+
+func (g *plastic) parseStatusChangeset(status string) int {
+	var csRegex = `STATUS\s+(?P<cs>[0-9]+?)\s`
+	cs, _ := strconv.Atoi(findNamedRegexMatch(csRegex, status)["cs"])
+	return cs
+}
+
+func (g *plastic) getHeadChangeset() int {
+	output := g.getCmCommandOutput("status", "--head", "--machinereadable")
+	var csRegex = `\bcs:(?P<cs>[0-9]+?)\s`
+	cs, _ := strconv.Atoi(findNamedRegexMatch(csRegex, output)["cs"])
+	return cs
+}
+
+func (g *plastic) getPlasticFileContents(folder, file string) string {
+	return strings.Trim(g.env.getFileContent(folder+"/"+file), " \r\n")
+}
+
+func (g *plastic) getSelector() string {
+	var ref string
+	selector := g.getPlasticFileContents(g.plasticWorkspaceFolder+"/.plastic", "plastic.selector")
+	// changeset
+	ref = g.parseChangesetSelector(selector)
+	if ref != "" {
+		return fmt.Sprintf("%s%s", g.props.getString(CommitIcon, "\uF417"), ref)
+	}
+	// fallback to label
+	ref = g.parseLabelSelector(selector)
+	if ref != "" {
+		return fmt.Sprintf("%s%s", g.props.getString(TagIcon, "\uF412"), ref)
+	}
+	// fallback to branch/smartbranch
+	ref = g.parseBranchSelector(selector)
+	if ref != "" {
+		ref = g.truncateBranch(ref)
+	}
+	return fmt.Sprintf("%s%s", g.props.getString(BranchIcon, "\uE0A0"), ref)
+}
+
+func (g *plastic) parseChangesetSelector(selector string) string {
+	var csRegex = `\bchangeset "(?P<cs>[0-9]+?)"`
+	return findNamedRegexMatch(csRegex, selector)["cs"]
+}
+
+func (g *plastic) parseLabelSelector(selector string) string {
+	var labelRegex = `label "(?P<label>[a-zA-Z0-9\-\_]+?)"`
+	return findNamedRegexMatch(labelRegex, selector)["label"]
+}
+
+func (g *plastic) parseBranchSelector(selector string) string {
+	var branchRegex = `branch "(?P<branch>[\/a-zA-Z0-9\-\_]+?)"`
+	return findNamedRegexMatch(branchRegex, selector)["branch"]
+}
+
+func (g *plastic) init(props properties, env environmentInfo) {
+	g.props = props
+	g.env = env
+}
+
+func (g *plastic) truncateBranch(branch string) string {
```

Comment:
Duplicate function

Reply:
True. Where would you refactor this function to? (I need a little guidance here)

Reply:
Good question, we could create an `scm` struct that has the functions (also status) and use struct inheritance in `git` and `plastic`

Reply:
This worked out well =) See [edb0fec](https://github.com/JanDeDobbeleer/oh-my-posh/pull/1392/commits/edb0fece458381486862e9e400feb375388231f9)